File: fix_routes.py
#!/usr/bin/env python3
import re

# Read the file
with open('server/routes.ts', 'r') as f:
    content = f.read()

# Fix all broken ApiResponse patterns
patterns_to_fix = [
    # Fix broken ApiResponse.validation with multiline
    (r'return ApiResponse\.validation\(res, "\s*\n\s*message: "([^"]+)"\s*\n\s*\}\);', 
     r'return res.status(400).json({\n        message: "\1"\n      });'),
    
    # Fix broken ApiResponse.validation with error field
    (r'return ApiResponse\.validation\(res, "\s*\n\s*message: "([^"]+)",?\s*\n\s*error: ([^\n]+)\s*\n\s*\}\);',
     r'return res.status(400).json({\n        message: "\1",\n        error: \2\n      });'),
     
    # Fix broken ApiResponse.internal
    (r'return ApiResponse\.internal\(res, "\s*\n\s*message: "([^"]+)"\s*\n\s*\}\);',
     r'return res.status(500).json({\n        message: "\1"\n      });'),
    
    # Fix broken ApiResponse.rateLimited
    (r'return ApiResponse\.rateLimited\(res, "\s*\n\s*message: "([^"]+)"\s*\n\s*\}\);',
     r'return res.status(429).json({\n        message: "\1"\n      });'),
     
    # Fix remaining ApiResponse calls
    (r'return ApiResponse\.notFound\(res, "([^"]+)"\);',
     r'return res.status(404).json({ message: "\1 not found" });'),
    (r'return ApiResponse\.forbidden\(res, "([^"]+)"\);',
     r'return res.status(403).json({ message: "\1" });'),
    (r'return ApiResponse\.unauthorized\(res, "([^"]+)"\);',
     r'return res.status(401).json({ message: "\1" });'),
    
    # Fix weird spacing in json objects
    (r'\.json\({(\s+)message:',
     r'.json({\n        message:'),
    
    # Fix broken closing braces
    (r'\}\);(\s*\})',
     r'});\1'),
]

# Apply all patterns
for pattern, replacement in patterns_to_fix:
    content = re.sub(pattern, replacement, content, flags=re.MULTILINE)

# Write the fixed content
with open('server/routes.ts', 'w') as f:
    f.write(content)

print("Fixed routes.ts file")

File: fix_routes.mjs
import fs from 'fs';

// Read the file
let content = fs.readFileSync('server/routes.ts', 'utf8');

// Fix all broken ApiResponse patterns
const fixPatterns = () => {
  // Remove any remaining ApiResponse calls 
  content = content.replace(/return ApiResponse\.\w+\([^;]*?\);?/g, (match) => {
    if (match.includes('notFound')) {
      return 'return res.status(404).json({ message: "Not found" });';
    } else if (match.includes('forbidden')) {
      return 'return res.status(403).json({ message: "Forbidden" });';
    } else if (match.includes('unauthorized')) {
      return 'return res.status(401).json({ message: "Unauthorized" });';
    } else if (match.includes('validation')) {
      return 'return res.status(400).json({ message: "Validation error" });';
    } else if (match.includes('rateLimited')) {
      return 'return res.status(429).json({ message: "Rate limited" });';
    } else if (match.includes('internal')) {
      return 'return res.status(500).json({ message: "Internal server error" });';
    }
    return match;
  });

  // Fix missing closing braces
  content = content.replace(/res\.status\((\d+)\)\.json\({ message: ([^}]+)}\);/g, 
    'res.status($1).json({ message: $2 });');
  
  // Fix broken json message patterns
  content = content.replace(/\.json\({\s+message: "/g, '.json({ message: "');
  
  // Fix double closing parentheses
  content = content.replace(/\}\);\);/g, '});');
  
  // Remove trailing req parameters from error responses
  content = content.replace(/\}\);, req\);/g, '});');
};

fixPatterns();

// Write the fixed content
fs.writeFileSync('server/routes.ts', content);

console.log('Fixed routes.ts file');